enterprise/server/saml/saml.go

```diff
@@ -88,6 +92,11 @@ func (a *SAMLAuthenticator) AuthenticatedHTTPContext(w http.ResponseWriter, r *h
 			return ctx
 		}
 	}
+
+	if slug := auth.GetCookie(r, slugCookie); err != nil && slug != "" {
+		return auth.AuthContextWithError(ctx, status.PermissionDeniedErrorf("Error getting service provider for slug %s: %s", slug, err.Error()))
+	}
```

Comment:
i think this might be clearer as if err == nil {
.... block ...
} else {
  if slug := auth.GetCookie(r, slugCookie); slug != "" {
    ... do something
  } 
}

fallback case()

Reply:
Done